### DISCUSSION TITLE
running-javascript-in-a-dotnet-app-with-javascriptengineswitcher/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#372] [Announcements]
# running-javascript-in-a-dotnet-app-with-javascriptengineswitcher/

In this post I describe how you can use the JavaScriptEngineSwitcher NuGet package to run JavaScript inside a .NET application

https://andrewlock.net/running-javascript-in-a-dotnet-app-with-javascriptengineswitcher/

**Comment by situssoft** [comment]
Are you aware of any mechanism which provides for debugging the embedded javascript which the .NET runs? I'm interested in providing a javascript scripting environment in my applicaition and I'd like to provide for debugging.

**Reply by lahma** [reply]
Here you can find couple of repositories related to debugging with Jint: 

* https://github.com/Jither/Jint.DebuggerExample
* https://github.com/Jither/Jint.DebugAdapter

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at April 20 2022, 09:39, in reply to [Marko Lahma](https://disqus.com/by/markolahma/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/372#discussioncomment-6063060)</em>

---

<p>Hey Marko! I'm not sure, so I just gave it a try (man I love this wrapper library!). unfortunately prism.js seems to use some regex values which cause a problem for Jint. I haven't completely narrowed it down yet, but tbh, I didn't try too hard, and just switched to Jurassic as that seemed to work.</p><p>IIRC, the docker and git ones are problematic for starters. e.g. the file from this link: <a href="https://prismjs.com/download.html#themes=prism&amp;languages=markup+css+clike+javascript+docker+git" rel="nofollow noopener" title="https://prismjs.com/download.html#themes=prism&amp;languages=markup+css+clike+javascript+docker+git">https://prismjs.com/download.html#themes=prism&amp;languages=markup+css+clike+javascript+docker+git</a></p>

[a comment by an automated account is not shown]

**Comment by krompaco** [comment]
Thanks for this! Use it pretty much straight off with Prism.js on my blog:
https://krompaco.nu/2023/02/matching-selenium-webdriver-chromedriver-nuget-version-with-machine/

**Comment by lahma** [comment]
Hey folks, Jint 3.0.0 has been released with almost full ES2023 level support (full generators support pending). A very different beast compared to v2 from performance standpoint too.

https://github.com/sebastienros/jint/releases/tag/v3.0.0
